cpp/src/groupby/sort/functors.hpp

```diff
@@ -40,12 +40,23 @@ struct store_result_functor {
                        sort::sort_groupby_helper& helper,
                        cudf::detail::result_cache& cache,
                        rmm::cuda_stream_view stream,
-                       rmm::mr::device_memory_resource* mr)
-    : helper(helper), cache(cache), values(values), stream(stream), mr(mr)
+                       rmm::mr::device_memory_resource* mr,
+                       sorted keys_are_sorted = sorted::NO)
```

Comment:
Would it be possible/preferable to set `sorted_values`/`grouped_values` as views to `values` in the constructor, when keys are sorted? To me it looks like this would replace multiple early exits (the fewer branches the better).
I think this would eliminate the early exit from `scan_result_functor::get_grouped_values` as well. Also eliminates the need for `is_presorted` AFAICT.

Reply:
Right, we can set `sorted_values`/`grouped_values` as views of existing data like `values` if input keys are presorted. When the input data is not sorted, however, creating and materializing `sorted_values`/`grouped_values` is necessary thus I'm not sure how the non-owning view will not work in this case. Setting up `sorted_values`/`grouped_values` in the constructor when presorted can avoid early exit branches but will cause redundant materialization.

Reply:
@karthikeyann any comments on your side are appreciated since you are the original author of this code.

Reply:
Ah, I mixed it up, as if `values` is a column and `sorted_values` is a view :\
There's still a way to do what I suggested, but AFAICT it's not better than the current code.